### PR TITLE
symtab: optimize, add test and fix linking issue

### PR DESCRIPTION
**Pull request by ycsin**
- Added a function to sanitize function names in the Python script
- Renamed the `start_addr` to `first_addr`
- Test symtab - somehow I forgot to add tests for it
- Added a new `Debug: Symtab` area in the maintainer list, LMK if this should just go into `Debug`
- Use linkonce (similar to `sw_isr_table`) to fix linking issue with external toolchain

[Doc preview](https://builds.zephyrproject.io/zephyr/pr/73333/docs/services/debugging/index.html)

### CODE REVIEW

**Review by cfriedt — DISMISSED** [review]
LGTM -

If unsure, common iterator bounds are `begin` and `end` (taken from C++), where `end` usually lies outside of the range so that `for (it = begin; it < end; ..)` works.

One thing I was curious about was whether the symbol table is sorted numerically by address. If it's sorted, then you can do a binary search on it based on the address, which is O(log(n)) (i.e. pretty good).

I haven't looked into too much detail to see how it's done though.

File: doc/services/debugging/mipi_stp_decoder.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ written to Stimulus Port registers maps directly to STP stream.
 This module can be used to perform on-chip decoding of the data stream. STP v2 is used.
 
 Usage
-#####
```
Comment (nashif) [inline doc/services/debugging/mipi_stp_decoder.rst:20]: that is unrelated change, isnt it?

Comment (ycsin) [reply doc/services/debugging/mipi_stp_decoder.rst:20]: Not really related to the original intent of this PR, but something seems off while I'm here:

Screenshot 2024-05-27 at 10 32 46 PM

I guess the `Usage` and `Limitations` probable should stay inside the MIPI documentation

Comment (ycsin) [reply doc/services/debugging/mipi_stp_decoder.rst:20]: Screenshot 2024-05-28 at 11 07 41 AM

What it would look like after this PR

Comment (nashif) [reply doc/services/debugging/mipi_stp_decoder.rst:20]: looks good

**Review by nashif — DISMISSED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by nashif — APPROVED** [review]
